packages/attestation-service/src/requestHandlers/get_attestation.ts

```diff
@@ -36,10 +36,47 @@ class GetAttestationRequestHandler {
     this.key = getAttestationKey(getRequest)
   }
 
-  async getAttestationRecord() {
-    return findAttestationByKey(this.key, {
+  async withAttestationAndSecurityCodeChecked(f: (attestation: AttestationModel) => any) {
+    const transaction = await sequelize!.transaction({
       logging: this.sequelizeLogger,
+      type: Transaction.TYPES.IMMEDIATE,
     })
+    try {
+      const attestation = await findAttestationByKey(this.key, {
+        transaction,
+        lock: Transaction.LOCK.UPDATE,
+      })
+
+      if (!attestation) {
+        throw new ErrorWithResponse('Attestation not found', 404)
+      }
+
+      // Check security code is supplied and attempts allow
+      if (!this.getRequest.securityCode) {
```

Comment:
I'm probably missing something, but does this mean that if the get request does not contain a security code, then the getRequest is approved? Doesn't this prove problematic?

Reply:
"Doesn't this prove problematic" is an understatement!  pls check the updated logic.

Reply:
yikes good catch!